crates/biome_js_analyze/src/semantic_analyzers/correctness/use_hook_at_top_level.rs

```diff
@@ -116,10 +282,11 @@ impl Rule for UseHookAtTopLevel {
         let options = ctx.options();
         let options = ReactExtensiveDependenciesOptions::new(options.clone());
 
-        let call = ctx.query();
+        let FunctionCall(call) = ctx.query();
         let hook_name_range = call.callee().ok()?.syntax().text_trimmed_range();
         if react_hook_configuration(call, &options.hooks_config).is_some() {
-            let model = ctx.model();
+            let model = ctx.semantic_services.model();
+            let early_returns = &ctx.early_returns;
```

Comment:
nit: I would create two functions to abstract the consumption of these two models:

```rs
let semantic_model = ctx.semantic_model();
let early_returns_model = ctx.early_returns_model();
```

Reply:
Sure.